### PR TITLE
Ak/total price

### PR DESCRIPTION
**Pull request by salci1990**

### CODE REVIEW

File: src/main/java/MainApp.java [hunk 1]
```diff
@@ -19,13 +29,64 @@ public void setUp() {
...
 
+    @Test
+    public void loginErrorTest() {
+        driver.get("http://automationpractice.com/index.php");
+        header.clickLoginLink();
+        signInPage.clickSignInLink();
+        assertThat(signInPage.isAlertOn());
+    }
+
+    @Test
```
Comment (bkita) [inline src/main/java/MainApp.java:58]: java conventions says that method should start from small letter!!!

File: src/main/java/MainApp.java [hunk 2]
```diff
@@ -19,13 +29,64 @@ public void setUp() {
         options.addArguments("disable-infobars");
         driver = new ChromeDriver(options);
         wait = new WebDriverWait(driver, 60);
+
+        signInPage = new SignInPage(driver);
+        header = new Header(driver);
+        indexPage = new IndexPage(driver);
+        signInPage = new SignInPage(driver);
+        header = new Header(driver);
```
Comment (bkita) [inline src/main/java/MainApp.java:40]: is it layer card or layer cart (t or d)?

File: src/main/java/MainApp.java [hunk 3]
```diff
@@ -19,13 +29,64 @@ public void setUp() {
...
+
+        indexPage.clickAddButton();
+
+        wait.until(ExpectedConditions.visibilityOf(layerCard.exitWindow()));
+        assertThat(layerCard.exitWindow().isDisplayed()).isTrue();
+        layerCard.exitWindowSuccess();
+
+        builder.moveToElement(indexPage.findElement().get(3)).build().perform();
+
+        indexPage.clickAddButton();
```
Comment (bkita) [inline src/main/java/MainApp.java:73]: please find a way to DO NOT use sleep

File: src/main/java/MainApp.java [hunk 4]
```diff
@@ -19,13 +29,64 @@ public void setUp() {
...
+        }
+        wait.until(ExpectedConditions.visibilityOf(layerCard.exitWindow()));
+        assertThat(layerCard.exitWindow().isDisplayed()).isTrue();
+        layerCard.exitWindowSuccess();
+
+        builder.moveToElement(header.moveToShopingCard()).build().perform();
+        wait.until(ExpectedConditions.visibilityOf(header.checkPriceCard()));
+        assertThat(header.checkPriceCard().isDisplayed()).isTrue();
+        header.openShoppingCard();
+
```
Comment (bkita) [inline src/main/java/MainApp.java:85]: this formatting should go to the page object, you can return formatted total and sum from page object

File: src/main/java/page_objects/Header.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+public class Header extends PageObject {
+
+    public Header(WebDriver driver) {
+        super(driver);
+    }
+
+    @FindBy(className = "login")
+    private WebElement loginLink;
+
+    @FindBy(partialLinkText = "Cart")
```
Comment (bkita) [inline src/main/java/page_objects/Header.java:16]: cart!

File: src/main/java/page_objects/Header.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    private WebElement shoppingCard;
+
+    @FindBy(id = "button_order_cart")
+    private WebElement pricesCard;
```
Comment (bkita) [inline src/main/java/page_objects/Header.java:20]: cart!

File: src/main/java/page_objects/IndexPage.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    private List<WebElement> listOfElements;
+
+    public WebElement exitWindowSuccess() {
+        return exitWindow;
+    }
+
+    public void clickAddButton() {
+        addButton.click();
+    }
+
```
Comment (bkita) [inline src/main/java/page_objects/IndexPage.java:31]: please rename method to addButton

File: src/main/java/page_objects/LayerCard.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package page_objects;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
```
Comment (bkita) [inline src/main/java/page_objects/LayerCard.java]: cart?

File: src/main/java/page_objects/LayerCard.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    @FindBy(className = "cross")
+    private WebElement exitWindow;
+
+    public WebElement exitWindow() {
+        return exitWindow;
+    }
+
+    public void exitWindowSuccess() {
```
Comment (bkita) [inline src/main/java/page_objects/LayerCard.java]: please rename method to closeLayerCart

File: src/main/java/page_objects/ShoppingCard.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
+package page_objects;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.text.DecimalFormat;
+
+public class ShoppingCard extends PageObject {
```
Comment (bkita) [inline src/main/java/page_objects/ShoppingCard.java]: cart

**Review by bkita — CHANGES_REQUESTED** [review]

File: src/main/java/MainApp.java (hunk 2 above)
Comment (salci1990) [reply src/main/java/MainApp.java:40]: done

File: src/main/java/MainApp.java (hunk 3 above)
Comment (salci1990) [reply src/main/java/MainApp.java:73]: done

File: src/main/java/MainApp.java (hunk 4 above)
Comment (salci1990) [reply src/main/java/MainApp.java:85]: done

File: src/main/java/page_objects/Header.java (hunk 5 above)
Comment (salci1990) [reply src/main/java/page_objects/Header.java:16]: done

File: src/main/java/page_objects/Header.java (hunk 6 above)
Comment (salci1990) [reply src/main/java/page_objects/Header.java:20]: done

File: src/main/java/page_objects/IndexPage.java (hunk 7 above)
Comment (salci1990) [reply src/main/java/page_objects/IndexPage.java:31]: done

File: src/main/java/page_objects/LayerCard.java (hunk 8 above)
Comment (salci1990) [reply src/main/java/page_objects/LayerCard.java]: done

File: src/main/java/page_objects/LayerCard.java (hunk 9 above)
Comment (salci1990) [reply src/main/java/page_objects/LayerCard.java]: done

File: src/main/java/page_objects/ShoppingCard.java (hunk 10 above)
Comment (salci1990) [reply src/main/java/page_objects/ShoppingCard.java]: done

**Pull request merged**